src/qt/askpassphrasedialog.cpp

```diff
@@ -152,14 +152,15 @@ void AskPassphraseDialog::accept()
         }
         } break;
     case Unlock:
-        if(!model->setWalletLocked(false, oldpass))
-        {
-            QMessageBox::critical(this, tr("Wallet unlock failed"),
-                                  tr("The passphrase entered for the wallet decryption was incorrect."));
-        }
-        else
-        {
-            QDialog::accept(); // Success
+        try {
```

Comment:
I think this is as ugly as calling UI from low level, only inverted.

One alternative is to catch the exception in `WalletImpl::unlock` and return an error.

Orthogonally `CCryptoKeyStore::Unlock` could also return an error instead of `assert(false)`.

Reply:
> I think this is as ugly as calling UI from low level, only inverted.

I don't think I understand why this ugly. Can you explain? Is the problem just that the code throws an exception instead of returning an error code?

Reply:
Thanks for clarifying. It sounds like either switching to a return code, or switching from `std::runtime_error` to a specific exception type would address your concerns. I agree these things would be improvements. I'm also ok with current PR, though.

Reply:
ignore my diatribe, but exceptions are just the cpp-ish way of goto. Especially with our wildcard catch blocks, this is hard to review and easy to break in the future. Less braindead programming languages like rust show that error handling can be done without exceptions or gotos.